legend-sdlc-server/src/test/java/org/finos/legend/sdlc/server/gitlab/api/docker/AbstractGitLabApiTest.java

```diff
@@ -159,51 +151,13 @@ protected static GitLabUserContext prepareGitLabMemberUserContext() throws Legen
 
     /**
      * Authenticates to GitLab and creates a test GitLabUserContext.
+     *
      * @param username the name of user for whom we create this context.
      * @param password the password of user for whom we create this context.
      */
     private static GitLabUserContext prepareGitLabUserContextHelper(String username, String password) throws LegendSDLCServerException
     {
-        GitLabMode gitLabMode = GitLabMode.UAT;
-        HttpServletRequest httpServletRequest = new TestHttpServletRequest();
-
-        TestGitLabSession session = new TestGitLabSession(username);
-        GitLabApi oauthGitLabApi;
-        Version version;
-
-        try
-        {
-            oauthGitLabApi = GitLabApi.oauth2Login(TEST_HOST_URL, username, password, null, null, true);
-            Assert.assertNotNull(oauthGitLabApi);
-            version = oauthGitLabApi.getVersion();
-        }
-        catch (GitLabApiException e)
-        {
-            StringBuilder builder = new StringBuilder("Error instantiating GitLabApi via OAuth2; response status: ").append(e.getHttpStatus());
-            String eMessage = e.getMessage();
-            if (eMessage != null)
-            {
-                builder.append("; error message: ").append(eMessage);
-            }
-            if (e.hasValidationErrors())
-            {
-                builder.append("; validation error(s): ").append(e.getValidationErrors());
-            }
-            throw new LegendSDLCServerException(builder.toString(), e);
-        }
-
-        String oauthToken = oauthGitLabApi.getAuthToken();
-        LOGGER.info("Retrieved access token: {}", oauthToken);
-        Assert.assertNotNull(version);
-
-        GitLabServerInfo gitLabServerInfo = GitLabServerInfo.newServerInfo(TEST_HOST_SCHEME, TEST_HOST_HOST, TEST_HOST_PORT);
-        GitLabAppInfo gitLabAppInfo = GitLabAppInfo.newAppInfo(gitLabServerInfo, null, null, null);
-        GitLabModeInfo gitLabModeInfo = GitLabModeInfo.newModeInfo(gitLabMode, gitLabAppInfo);
-
-        session.setAccessToken(oauthToken);
-        session.setModeInfo(gitLabModeInfo);
-        LegendSDLCWebFilter.setSessionAttributeOnServletRequest(httpServletRequest, session);
-        return new GitLabUserContext(httpServletRequest, null);
+        return GitLabApiTestSetupUtil.prepareGitLabUserContextHelper(username, password, TEST_HOST_URL, TEST_HOST_SCHEME, TEST_HOST_HOST, TEST_HOST_PORT, LOGGER);
```

Comment:
It's best not to pass this class's logger to somewhere else. That can lead to confusion on where log messages are coming from.

Reply:
Fixed!